infra/src/main/kotlin/fi/fta/geoviite/infra/switchLibrary/SwitchLibraryService.kt

```diff
@@ -11,23 +11,18 @@ class SwitchLibraryService(
     private val switchStructureDao: SwitchStructureDao,
     private val switchOwnerDao: SwitchOwnerDao,
 ) {
+    private val structures: List<SwitchStructure> by lazy {
+        switchStructureDao.fetchSwitchStructures()
+    }
+
     private val structuresById: Map<IntId<SwitchStructure>, SwitchStructure> by lazy {
-        switchStructureDao.fetchSwitchStructures().associateBy { switchStructure ->
+        structures.associateBy { switchStructure ->
             switchStructure.id as IntId
         }
     }
```

Comment:
Kakutus hoidetaan nyt kokonaan Service-puolella. Lisäksi täältä tiedostosta poistettu turhaa koodia ja korjattu kakut ohittavia DAO-kutsuja käyttämään näitä kakkuja